ux.symfony.com/src/Service/EmojiCollection.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace App\Service;
+
+use IteratorAggregate;
+use Traversable;
+
+/**
+ * Collection of emojis used in demos.
+ *
+ * @implements IteratorAggregate<string>
+ *
+ * @internal
+ */
+final class EmojiCollection implements IteratorAggregate, \Countable
+{
+    private array $emojis;
+
+    public function __construct(array $emojis = [])
+    {
+        $this->emojis = $emojis ?: $this->loadEmojis();
```

Comment:
is it "ok" to do it in constructor ?
I know its memory array, but to have better code, perhaps compute this outside of constructor ?

Reply:
In real life, with a service, you would probably do it in a dedicated runtime getter, indeed. But in order to do short demo and showcase the code in the page, we often have to trade some "best practices" for better readability. 

As shown in the page, the goal is really  here to insist on the LiveProp / LiveAction so i'd like to keep the rest of the code minimal.

WDYT ?